cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -985,11 +986,17 @@ impl<M: ABIMachineSpec> ABICallee for ABICalleeImpl<M> {
     fn gen_epilogue(&self) -> Vec<M::I> {
         let mut insts = vec![];
 
+        let mut fixed_frame_storage_size = 0;
+        if !self.call_conv.extends_baldrdash() {
+            fixed_frame_storage_size += self.frame_size();
+        }
```

Comment:
```suggestion
        let fixed_frame_storage_size = if !self.call_conv.extends_baldrdash() {
            self.frame_size()
        } else {
            0
        };
```

Also why this condition?

Reply:
If you look at the code in gen_prologue above, it passes a fixed_frame_storage_size of 0 to gen_clobber_save with the baldrdash calling convention.  For consistency, we need to do the same for gen_clobber_restore.

Likewise, I followed the same coding convention with the "mut fixed_frame_storage_size" as is currently in place in gen_prologue.  Given that any future changes to the computation of the size will need to be kept in sync between gen_prologue and gen_epilogue, it seems preferable to me to also have the code itself as similar as possible.

Reply:
I think it would be better to simply store the `fixed_frame_storage_size` in `self` -- we get a mutable `self` in `gen_prologue` so we should be able to write it there. This avoids the duplication of logic and potential for mismatches...

Reply:
OK, updated the patch accordingly.